src/us/kbase/workspace/kbase/IdentifierUtils.java

```diff
@@ -267,12 +264,21 @@ private static ObjectIdentifier buildObjectIdentifier(
 					"Error on ObjectSpecification #"
 					+ objcount + ": " + e.getLocalizedMessage(), e);
 		}
-		if (paths == null && refchain == null) {
+		final boolean searchDAG = longToBoolean(o.getFindReferencePath());
+		if (paths == null && refchain == null && !searchDAG) {
 			return oi;
 		} else if (paths == null) {
-			return new ObjectIDWithRefPath(oi, refchain);
+			if (searchDAG) {
+				return new ObjectIDWithRefPath(oi);
+			} else {
+				return new ObjectIDWithRefPath(oi, refchain);
```

Comment:
What happens if both "find_reference_path" and one of "obj_path/obj_ref_path/to_obj_path/to_obj_ref_path" are defined. Looks like we silently disregard latter option, right?